### PR TITLE
luhn: add new exercise

### PR DESCRIPTION
**Pull request by filiptoskovic**

### CODE REVIEW

**Comment by filiptoskovic** [comment]
There is a problem with test `a simple valid SIN that becomes invalid if reversed`. It is `true` for input value `59`. Value tor 59 is false ( 5 + 9 * 2 - 9 = 14)  and expected value is true. Value of test is in collision with test name. Can this be an error in canonical or I am completely wrong about this exercise?

**Comment by rpottsoh** [comment]
I don't see the example solution reversing anywhere.  I am also looking at the csharp example and can obviously see a reversal taking place.  Refer to the csharp example if you like.  At the moment all test are passing.  The test suite looks fine to me.

**Comment by filiptoskovic** [comment]
I have refactored example and changed test `a simple valid SIN that becomes invalid if reversed`. I think that I didn't understood this exercise at all. In C# solutions calculate check digit and a lot of different things, but I don't see the point of that. Also, first test said: 'one digit is invalid' and second test said: 'one zero is invalid'. Zero is a digit, what is the point of that? 
Fourth test said: 'a_simple_valid_SIN_that_becomes_invalid_if_reversed' and my understanding is that input is valid and will be invalid if reversed. But input is invalid and will be valid if reversed, or input is string after reversing?

File: exercises/luhn/uLuhnExample.pas [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+{ TLuhn }
+
+class function TLuhn.Valid(AInp: string): boolean;
+var
+  s : string;
+  i, R, P: int64;
+begin
+  R := 0;
+  s := ReverseString(TRegEx.Replace(AInp, ' ', ''));
```
Comment (rpottsoh) [inline exercises/luhn/uLuhnExample.pas]: I think this will accomplish the same thing.
```pascal
s := ReverseString(AInp.Replace(' ', ''));
```

Comment (rpottsoh) [reply exercises/luhn/uLuhnExample.pas]: The luhn equation states that we must move from right to left through the string.  I think this is why other solutions are reversing the input string.  Be sure to review https://en.wikipedia.org/wiki/Luhn_algorithm.

Don't worry about what the case descriptions say so much.  Just input the prescribed input and expect the expected response.

**Comment by filiptoskovic** [comment]
yes, you are right.  There were `s := ReverseString(TRegEx.Replace(AInp, '\D', ''));` before, and I just replaces non digit with space. There is no need for regex. I have looked to Luhn algorithm (We use it in Serbia for Healh Insurance Number) but still don't see why every solution in C# have function for check digit if we test validity only. I'm sure that I'm missing something.

**Comment by rpottsoh** [comment]
I think just make sure that the check digit is not doubled.

**Comment by rpottsoh** [comment]
@filiptoskovic are planning to make any more changes.  If not I will go ahead and merge.  We can always make changes to the example solution later if necessary.  I think the test suite is OK.
